guide/src/mocking_lros.md

```diff
@@ -0,0 +1,183 @@
+<!-- 
+Copyright 2025 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# How to write tests for long-running operations
+
+The Google Cloud client libraries for Rust have helpers that simplify
+interaction with long-running operations (henceforth, LROs).
+
+Simulating the behavior of LROs in tests involves reverse-engineering some of
+those helpers. This guide shows how to do that.
+
+## Prerequisites
+
+This guide assumes you are familiar with the previous chapters:
+
+- [Working with long-running operations](working_with_long_running_operations.md)
+- [How to write tests using a client](mock_a_client.md)
+
+## Tests for automatic polling
+
+Let's say our application code awaits `lro::Poller::until_done()` (in previous
+sections, we called this automatic polling).
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:auto-fn}}
+```
+
+Note that our application only cares about the final result of the LRO. We do
+not need to test how it handles intermediate results from polling the LRO. Our
+tests can simply return the final result of the LRO from the mock.
+
+### Creating the `longrunning::model::Operation`
+
+Let's say we want our call to result in the following response.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:expected-response}}
+```
+
+You may have noticed that the stub returns a `longrunning::model::Operation`,
+not a `BatchRecognizeResponse`. We need to pack our desired response into the
+`Operation::result`.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:finished-op}}
+```
+
+Note also that we set the `done` field to `true`. This indicates to the `Poller`
+that the operation has completed, thus ending the polling loop.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:set-done-true}}
+```
+
+### Test code
+
+Now let's get to the test code.
```

Comment:
```suggestion
With the mocks ready, we can write the corresponding test.
```

?? Seems odd to say that we are only how looking at tests, mocks are kind of part of the tests?

Reply:
Reworded to "Now we are ready to write our test."